x-pack/solutions/observability/plugins/synthetics/server/cases/suggestion.ts

```diff
@@ -0,0 +1,236 @@
+                  },
+                  {
+                    terms: {
+                      'summary.final_attempt': [true],
+                    },
+                  },
+                ],
+              },
+            },
+            aggs: {
+              by_monitor: {
+                terms: {
+                  field: 'observer.name',
+                  // TODO: TBD
+                  // it limits suggestions to the top 5 monitors
+                  size: 5,
+                },
+                aggs: {
+                  latest_run: {
+                    top_hits: {
+                      sort: [
+                        {
+                          '@timestamp': {
+                            order: 'desc',
+                          },
+                        },
+                      ],
+                      size: 1,
+                      _source: {
+                        includes: [
+                          'monitor.name',
+                          'monitor.type',
+                          'monitor.status',
+                          'status',
+                          'observer.geo.name',
+                          'observer.name',
+                          'service.name',
+                          'monitor.id',
+                          'config_id',
+                          'url.full',
+                          '@timestamp',
+                          'meta.space_id',
+                          'type',
+                        ],
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          });
+
+          const monitors = (results.aggregations as SyntheticsAggregationsResponse['aggregations'])
+            .by_monitor.buckets;
+
+          if (monitors.length === 0) {
+            logger.debug(`No Synthetics monitors found for service ${serviceNames.join(', ')}`);
+            return { suggestions: [] };
+          }
+
+          const savedObjectsAttrHash: Record<string, EncryptedSyntheticsMonitorAttributes> = {};
+          const bulkGetRequests: Array<{ id: string; type: string }> = [];
+
+          for (const mon of monitors) {
+            bulkGetRequests.push({
+              id: mon.latest_run.hits.hits[0]._source.config_id,
+              type: syntheticsMonitorSavedObjectType,
+            });
+          }
+
+          const monitorsSavedObject = await scopedSavedObjectsClient.bulkGet<{
+            monitor: Record<string, any>;
+          }>(bulkGetRequests);
+
+          if (bulkGetRequests.length === 0) {
+            logger.debug(`No Synthetics SavedObjects found for the related tests runs`);
+            return { suggestions: [] };
+          }
+
+          for (const savedObj of monitorsSavedObject.saved_objects) {
+            if (!savedObj.error && savedObj.id) {
+              savedObjectsAttrHash[savedObj.id] =
+                savedObj.attributes as unknown as EncryptedSyntheticsMonitorAttributes;
```

Comment:
Why do we need to cast it to unknown?

Reply:
Because it is a generic `attribute: T;`

Reply:
Thanks! I'm adding `monitor` unnecessarily to the payload.